software/utils/debug/debug_cli.py

```diff
@@ -816,42 +878,23 @@ def do_eeprom(self, line):
             return
 
 
-def auto_select_port():
-    ports = detect_stm32_ports()
-    if not ports:
-        red(
-            "Could not auto-detect serial port; platformio device list did not "
-            "yield any STM32 devices."
-        )
-        return None
-    if len(ports) > 1:
-        red(
-            "Could not auto-detect serial port; platformio device list "
-            f"yielded multiple STM32 devices: {', '.join(ports)}.  "
-            "Choose port explicitly with --port."
-        )
-        return None
-    return ports[0]
-
-
 def main():
     terminal_parser = argparse.ArgumentParser()
     terminal_parser.add_argument(
-        "--port",
-        "-p",
+        "--device",
+        "-d",
         type=str,
-        help="Serial port device is connected to, e.g. /dev/ttyACM0."
-        " If unspecified, we try to auto-detect the port.",
+        help="Device to connect to: serial, e.g. /dev/ttyACM0, or alias from manifest, or 'auto'.",
```

Comment:
Could we make 'auto' the default behavior, when no -d option is specified?

Reply:
It is possible, but may end up creating confusion for the more advanced user. At this layer, the debug interface must be capable of starting up without logging into anything, even if a device is available, simply to be used for using the `device` command, which `controller.sh` needs to fetch the manifest data. We could flip the logic on its head and add another option or something like `-d null` to not explicitly avoid connecting, but that would mean the CLI option would behave differently from the `device connect` command, which.... not sure that's nice to have 2 opposite paradigms. However, with the current behavior, it is simple enough to simply add your local device to the manifest and then add an `SN=` variable to the environment. You could even have `SN=/dev/ttyAMC0` and I think that would just blanket work for any device even without adding it to manifest. Would either of these solutions be ok?